Automattic/jetpack
# Protect: Parse data from the initial state

This PR adds a custom hook that reads and parses data from the initial state and make it available to be used by components.

It's a hook because it started out using `useContext()` but then I realized it was not needed. Even though it is not technically a hook, it looks good. But we could refactor it as a simple export/import if you prefer.

#### Changes proposed in this Pull Request:

* Parse data from initial state and use it in the UI

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Make sure to add to your wp-config:
```
define( 'JETPACK_PROTECT_DEV__BYPASS_CACHE', true );
define( 'JETPACK_PROTECT_DEV__API_RESPONSE_TYPE', 'complete_green' );
```

Visit Protect and confirm the information matches your list of installed plugins and themes.

Play with the response_type to see different results. Possible values are:
* complete_green
* complete
* incomplete_green
* incomplete
* empty -> will return the status of when the first check was not performed yet

complete means all installed plugins were checked
incomplete means some installed plugins were not checked
